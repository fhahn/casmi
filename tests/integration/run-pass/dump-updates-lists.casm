// cmdline "-u"

function a : -> Int initially { 1 }
function b : -> List(Int) initially { [] }
function c : Int * List(Int) -> List(Int)

init main

rule main = {
    a := a + 1
    b := cons(a, b)
    c(a, b) := b
    if a = 5 then
        program(self) := undef
}
