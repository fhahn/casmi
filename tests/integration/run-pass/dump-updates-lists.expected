{ a = 2, b = [ 1 ], c(1, [  ]) = [  ] }
{ a = 3, b = [ 2, 1 ], c(2, [ 1 ]) = [ 1 ] }
{ a = 4, b = [ 3, 2, 1 ], c(3, [ 2, 1 ]) = [ 2, 1 ] }
{ a = 5, b = [ 4, 3, 2, 1 ], c(4, [ 3, 2, 1 ]) = [ 3, 2, 1 ] }
{ program(self) = undef, a = 6, b = [ 5, 4, 3, 2, 1 ], c(5, [ 4, 3, 2, 1 ]) = [ 4, 3, 2, 1 ] }
5 steps later...
